master/buildbot/reporters/mail.py

```diff
@@ -43,6 +43,11 @@
 from buildbot.reporters.message import MessageFormatter as DefaultMessageFormatter
 from buildbot.util import service
 
+if PY3:
```

Comment:
If String means different thing in both python, I would suggest not to use that name.

from buildbot.util import BytesIO
from buildbot.util import UnicodeIO

This is for me the most unambiguous way of dealing with the problem.
Sorry to be a bit tough but I think those bytes vs string vs unicode are very error prone when a bunch of python2 coders want to write py3 compatible code.